pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -260,9 +280,13 @@ func (l *podNIC) PlugPhase1() error {
 			return errors.CreateCriticalNetworkError(err)
 		}
 
-		if err := bindMechanism.setCachedDhcpConfig(); err != nil {
-			log.Log.Reason(err).Error("failed to save dhcpConfig configuration")
-			return errors.CreateCriticalNetworkError(err)
+		dhcpConfigurator := l.dhcpConfigurator
+		if dhcpConfigurator != nil {
+			dhcpConfig := bindMechanism.generateDhcpConfig()
```

Comment:
dhcpConfig is generated the same way independently if its for bridge or masquerade, we can just move it here so we don't pollute the bindings with DHCP.

Reply:
While I agree the method's name is pretty poor (generate ... it is a simple getter), we can't move it to the pod nic, because the pod nic doesn't have the slightest idea of what the dhcpConfig is. That entry gets created in the constructor, then heavily mutated by the `discoverPodInterface` method. 

I'd go as far as to say the whole purpose of the `discoverPodInterface...` method is to mutade the `dhcpConfig` member of the corresponding `bindMechanism` implementation.

One alternative I can think of here is to make the `discover...` method return the `dhcpConfig` struct. But that would require plenty more changes to the bind mechanism implementations, because their `preparePodInterface` use the dhcp config struct for their own means:
 - masquerade: https://github.com/kubevirt/kubevirt/blob/5d34e64592c7dfc3e863da559d6d328c5bd65971/pkg/virt-launcher/virtwrap/network/podinterface.go#L1209
 - bridge: https://github.com/kubevirt/kubevirt/blob/5d34e64592c7dfc3e863da559d6d328c5bd65971/pkg/virt-launcher/virtwrap/network/podinterface.go#L509
 
 I'm sure there are more usages ...